diff -urNp -x '*.orig' fio-3.23.org/configure fio-3.23/configure
--- fio-3.23.org/configure	2020-09-06 18:45:11.000000000 +0200
+++ fio-3.23/configure	2021-10-09 23:26:34.042251668 +0200
@@ -1325,7 +1325,8 @@ int main(int argc, char **argv)
   return 0;
 }
 EOF
-if compile_prog "" "" "guasi"; then
+if compile_prog "" "-lguasi" "guasi"; then
+  LIBS="$LIBS -lguasi"
   guasi="yes"
 fi
 print_config "GUASI" "$guasi"
